compiler/rustc_session/src/options.rs

```diff
@@ -681,10 +686,9 @@ fn build_options<O: Default>(
                         ),
                     }
                 }
-                if let Some(tmod) = *tmod
-                    && let Some(value) = value
-                {
-                    target_modifiers.insert(tmod, value.to_string());
+                if let Some(tmod) = *tmod {
+                    let v = if let Some(v) = value { v.to_string() } else { String::new() };
```

Comment:
would `value.map_or_default(ToOwned::to_owned)` work here?

Reply:
Changed to `map_or(...)`, looks like `map_or_default` is not yet merged for Option.